### PR TITLE
AwesomeBooks

### PR DESCRIPTION
**Pull request by davidolanrewaju**
# 📚AwesomeBooks
>This PR was created to give details on the work done in this branch. In this branch, the logic to add a book dynamically when the add button is clicked and to also remove any of the added books in our book list was implemented

## 🛠Built With:
- HTML
- CSS
- JS
- Github Flows
- Linters(webhint, eslint, stylelint, lighthouse)

## ♻Implemented Features:
- [x] Add books to the book list dynamically when the add button is clicked
- [x] Remove a book from the book list dynamically

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            <li class="nav-link">
+                <a href="./index.html">List</a>
+            </li>
+            <li class="nav-link">
+                <a href="./add-books.html">Add Books</a>
+            </li>
+            <li class="nav-link">
+                <a href="./contact.html">Contact</a>
+            </li>
+        </ul> -->
```
Comment (AnselemOdims) [inline index.html]: - [ ] Please kindly get rid of any commented code as it pollutes the codebase

File: remove.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+// define a function to remove a book item
+function removeBook(event) {
+  // get the clicked remove button
+  const removeBtn = event.target;
+  // get the closest ancestor element with the 'book-list' class
+  const bookItem = removeBtn.closest('.book-list');
+  // remove the book item from the DOM
+  bookItem.remove();
+}
```
Comment (AnselemOdims) [inline remove.js]: - [ ] At the moment, this remove functionality will only remove the book from the screen and not from the storage. Please kindly ensure it removes the right book from the screen and storage.

File: storage.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+const authorInput = document.getElementById('author');
+
+// Save the form data to local storage whenever any input changes
+function saveFormData() {
+  const formData = {
+    title: titleInput.value,
+    author: authorInput.value,
+  };
+  localStorage.setItem('form-data', JSON.stringify(formData));
+}
```
Comment (AnselemOdims) [inline storage.js]: - [ ] At the moment, you are only saving the form data and not the list of books in your localStorage, so when a hard reload is done on the page, all added books disappear. Please kindly ensure the books are added to the local storage when created.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @dave-prog , 
You have been doing a tremendous job so far :ok_hand:
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
No linter errors :heavy_check_mark:

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by mwanawabangona — APPROVED** [review]
Hi @dave-prog 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**